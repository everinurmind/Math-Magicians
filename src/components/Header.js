import '../styles/Header.css';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header className="site-header">
    <h1 className="site-title">Math Magicians</h1>
    <nav className="main-navigation">
      <ul className="menu">
        <li><NavLink to="/" className="menu-item" exact>Home</NavLink></li>
        {' | '}
        <li><NavLink to="/calculator" className="menu-item">Calculator</NavLink></li>
        {' | '}
        <li><NavLink to="/quote" className="menu-item">Quote</NavLink></li>
      </ul>
    </nav>
  </header>
);

export default Header;
