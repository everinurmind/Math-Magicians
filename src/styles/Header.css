.site-header {
  background-color: #6b7a8f;
  color: #fff;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px;
  width: 100%;
}

.site-title {
  font-size: 36px;
  margin: 0;
  padding: 0 15px;
  text-shadow: 2px 2px #000;
}

.main-navigation {
  display: flex;
}

.menu {
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0 15px;
  gap: 25px;
}

.menu-item {
  color: #f0c808;
  font-weight: bold;
  text-decoration: none;
}

.menu-item:hover {
  color: lightpink;
}

.menu-item.active {
  color: lightcoral;
}
