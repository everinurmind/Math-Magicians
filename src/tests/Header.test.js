import renderer from 'react-test-renderer';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Header from '../components/Header';

test('should render Header.js', () => {
  const text = renderer.create(
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Header />} />
      </Routes>
    </BrowserRouter>,
  );

  expect(text.toJSON()).toMatchSnapshot();
});
