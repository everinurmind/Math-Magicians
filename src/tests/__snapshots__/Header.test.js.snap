// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header.js 1`] = `
<header
  className="site-header"
>
  <h1
    className="site-title"
  >
    Math Magicians
  </h1>
  <nav
    className="main-navigation"
  >
    <ul
      className="menu"
    >
      <li>
        <a
          aria-current="page"
          className="menu-item active"
          exact={true}
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
       | 
      <li>
        <a
          className="menu-item"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
       | 
      <li>
        <a
          className="menu-item"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
